lib/board.rb

```diff
@@ -1,77 +1,82 @@
-    valid_range.each do |coord|
-      board_hash[coord] = Cell.new(coord)
-    end
+    valid_range.each { |coords| board_hash[coords] = Cell.new(coords) }
     board_hash
   end
 
-  def valid_coordinate?(coord)
-    @cells.any? { |key, value| key == coord }
+  def valid_coordinate?(coords)
+    @cells.any? { |key, value| key == coords }
+  end
+
+  def valid_horizontal_placement?(coords, ship)
+    coords.all? { |coord| coord[0] == coords.first[0] } &&
+      (coords.first..coords.last).to_a == coords &&
+      coords.length == ship.length
+  end
+
+  def valid_vertical_placement?(coords,ship)
+    letters_in_coords = coords.map { |coord| coord[0] }
+    letters_in_coords_using_range = (coords.first[0]..coords.last[0]).to_a
+
+    coords.all? { |coord| coord[1] == coords.first[1] } &&
+     letters_in_coords == letters_in_coords_using_range
+  end
+
+  def out_of_bounds_or_has_ship_already?(coords)
+    coords.any? { |coord| !valid_coordinate?(coord) } ||
+      coords.any? { |coord| !@cells[coord].empty?}
   end
+  
 
-  def valid_placement?(ship, coord)
+  def valid_placement?(ship, coords)
     valid = false
-    #Rationale: If none of the horizontal || vertical rules apply, then not valid
 
-    #rules to place horizontally: letter same, sequintial nums == ship length
-    valid = true if coord.all? { |item| item[0] == coord.first[0] } &&
-                     (coord.first[1]..coord.last[1]).to_a.length == ship.length
+    valid = true if valid_horizontal_placement?(coords, ship)
 
-    #rules to place vertically: number same,  sequintial letters == ship.length
-    valid = true if coord.all? { |item| item[1] == coord.first[1] } &&
-                    (coord.first[0]..coord.last[0]).to_a.length == ship.length
+    valid = true if valid_vertical_placement?(coords,ship)
 
-    #all cells must be within the boards range and not contain ship
-    valid = false if coord.any? { |item| !valid_coordinate?(item) } ||
-                     coord.any? { |item| !@cells[item].empty?}
+    valid = false if out_of_bounds_or_has_ship_already?(coords)
 
-                     #check for odd middle coordinate
-    if coord.length == 3
-      valid = false if coord[0] > coord[1] || coord[0] == coord[1] || coord[1] == coord[2] || coord[2] < coord[1]
-    end
     valid
-    #sal - break this into methods
+ 
   end
 
-  def place(ship, coord)
-    is_valid = valid_placement?(ship, coord)
+  def place(ship, coords)
+    is_valid = valid_placement?(ship, coords)
 
-    if is_valid
-      coord.each do |item|
-        cells[item].place_ship(ship)
-      end
-    end
+    coords.each { |coord| cells[coord].place_ship(ship) } if is_valid
+ 
     is_valid
   end
 
   def render(show_ship = false)
-    board = "  1 2 3 4"
-    letter_arr = ['A', 'B', 'C', 'D']
-    @cells.each_with_index do |(coord, cell), i|
-      if i % 4  == 0
-        board << " \n"
-        board << letter_arr[0]
-        letter_arr.shift
-      end
-
-      if (i % 4) + 1 != 0
```

Comment:
I removed this logic because we actually want every iteration to have a space placed